### PR TITLE
Add acme_requestor metadata

### PR DESCRIPTION
**Pull request by inahga**
Closes https://github.com/peppelinux/draft-demarco-acme-openid-federation/issues/24

I think this also relates to https://github.com/peppelinux/draft-demarco-acme-openid-federation/issues/21.

### CODE REVIEW

File: draft-demarco-acme-openid-federation.md [hunk 1]
```diff
@@ -209,13 +208,13 @@ an X.509 Certificate.
...
+   Configuration, according to [Requestor Metadata](#requestor-metadata).
 
-5. The Certificate Issuer MAY be a Leaf, in these cases a specific Trust Mark
+5. The Issuer MUST publish the metadata type `acme_provider` in its Entity
+   Configuration, according to [Issuer Metadata](#issuer-metadata).
+
+6. The Certificate Issuer MAY be a Leaf, in these cases a specific Trust Mark
    enabling the issuance of X.509 Certificates within the federation MAY be
    issued by the Trust Anchor, or on behalf of it by an allowed Trust Mark
    issuer as configured in the federation. When used, the Trust Mark MUST be
```
Comment (peppelinux) [inline draft-demarco-acme-openid-federation.md:220]:
```suggestion
   Issuer as configured in the federation. When used, the Trust Mark MUST be
```

now in openidfed it is a defined term.

File: draft-demarco-acme-openid-federation.md [hunk 2]
```diff
@@ -261,7 +260,9 @@ Requestor is part of the federation, these are listed below:
       Discovery](https://openid.net/specs/openid-federation-1_0.html#section-8)
       to obtain the Trust Chain related to the Requestor.
 
-## Metadata
+## Entity Configuration Metadata
+
```
Comment (peppelinux) [inline draft-demarco-acme-openid-federation.md:264]:
```suggestion

This section contains the configuration for the Entities metadata, enabling the interoperability between the Issuer and the Requestor."
  

```

File: draft-demarco-acme-openid-federation.md [hunk 3]
```diff
@@ -304,11 +305,69 @@ the `acme_provider` metadata:
...
+        "iss": "https://requestor.example.com",
+        "sub": "https://requestor.example.com",
+        "iat": 1618410883
+      }
+    }
+  }
 }
 ~~~~
 
+After certificate issuance, the Requestor MAY remove the `acme_requestor`
```
Comment (peppelinux) [inline draft-demarco-acme-openid-federation.md:368]:
```suggestion
After X.509 Certificate issuance, the Requestor MAY remove the `acme_requestor`
```

Comment (peppelinux) [reply draft-demarco-acme-openid-federation.md:368]: what if the requestor does not remove those jwks?

be the issuer able to inspect if those requestor's jwks was already issued in the form of x.509 certificates and if not expired yet no action, or it should be mandated to reissue the certificates?

the key element is that it's up to the requestor specifying the key used for the x.509 request, this makes the presence of old/already issued public key impactless for the issuer

Comment (inahga) [reply draft-demarco-acme-openid-federation.md]: I will elaborate on this section a bit.

**Review by peppelinux — APPROVED** [review]

**Pull request merged**